flink-core/src/main/java/org/apache/flink/api/common/io/GenericCsvInputFormat.java

```diff
@@ -106,6 +106,11 @@ protected GenericCsvInputFormat() {
 	protected GenericCsvInputFormat(Path filePath) {
 		super(filePath);
 	}
+
+	protected GenericCsvInputFormat(Path filePath, Charset charset) {
+		super(filePath);
+		this.charset = charset != null ? charset : Charset.forName("UTF-8");
```

Comment:
Would this be better as `this.charset = Preconditions.checkNotNull(charset);` since the user should use `GenericCsvInputFormat(Path)` when using the default charset?